src/main/java/appeng/fluids/parts/PartFluidStorageBus.java

```diff
@@ -0,0 +1,508 @@
+import appeng.util.prioritylist.FuzzyPriorityList;
+import appeng.util.prioritylist.PrecisePriorityList;
+
+
+/**
+ * @author BrockWS
+ * @version rv6 - 22/05/2018
+ * @since rv6 22/05/2018
+ */
+public class PartFluidStorageBus extends PartSharedStorageBus implements IMEMonitorHandlerReceiver<IAEFluidStack>
+{
+	public static final ResourceLocation MODEL_BASE = new ResourceLocation( AppEng.MOD_ID, "part/fluid_storage_bus_base" );
+	@PartModels
+	public static final IPartModel MODELS_OFF = new PartModel( MODEL_BASE, new ResourceLocation( AppEng.MOD_ID, "part/fluid_storage_bus_off" ) );
+	@PartModels
+	public static final IPartModel MODELS_ON = new PartModel( MODEL_BASE, new ResourceLocation( AppEng.MOD_ID, "part/fluid_storage_bus_on" ) );
+	@PartModels
+	public static final IPartModel MODELS_HAS_CHANNEL = new PartModel( MODEL_BASE, new ResourceLocation( AppEng.MOD_ID, "part/fluid_storage_bus_has_channel" ) );
+
+	private final IActionSource source;
+	private final AppEngInternalAEInventory config = new AppEngInternalAEInventory( this, 63 );
+	private boolean cached = false;
+	private ITickingMonitor monitor = null;
+	private MEInventoryHandler<IAEFluidStack> handler = null;
+	private int handlerHash = 0;
+	private byte resetCacheLogic = 0;
+
+	public PartFluidStorageBus( ItemStack is )
+	{
+		super( is );
+		this.getConfigManager().registerSetting( Settings.ACCESS, AccessRestriction.READ_WRITE );
+		this.getConfigManager().registerSetting( Settings.FUZZY_MODE, FuzzyMode.IGNORE_ALL );
+		this.getConfigManager().registerSetting( Settings.STORAGE_FILTER, StorageFilter.EXTRACTABLE_ONLY );
+		this.source = new MachineSource( this );
+	}
+
+	private IMEInventory<IAEFluidStack> getInventoryWrapper( TileEntity target )
+	{
+		EnumFacing targetSide = this.getSide().getFacing().getOpposite();
+		// Prioritize a handler to directly link to another ME network
+		IStorageMonitorableAccessor accessor = target.getCapability( Capabilities.STORAGE_MONITORABLE_ACCESSOR, targetSide );
+		if( accessor != null )
+		{
+			IStorageMonitorable inventory = accessor.getInventory( this.source );
+			if( inventory != null )
+			{
+				return inventory.getInventory( AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) );
+			}
+
+			// So this could / can be a design decision. If the tile does support our custom capability,
+			// but it does not return an inventory for the action source, we do NOT fall back to using
+			// IItemHandler's, as that might circumvent the security setings, and might also cause
+			// performance issues.
+			return null;
+		}
+
+		// Check via cap for IItemHandler
+		IFluidHandler handlerExt = target.getCapability( CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, targetSide );
+		if( handlerExt != null )
+		{
+			return new FluidHandlerAdapter( handlerExt, this );
+		}
+
+		return null;
+	}
+
+	@Override
+	public TickRateModulation tickingRequest( IGridNode node, int TicksSinceLastCall )
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00117)